### PR TITLE
Bump go and ginkgo versions [dev]

### PR DESCRIPTION
**Pull request by gururajsh**
This PR

- Bumps the go version to 1.22.2
- Bumps ginkgo version to v2
- Adds github actions workflow to deploy plugins app.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    name: Running Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.18.3'
```
Comment (joaopapereira) [inline .github/workflows/test.yml:25]:
```suggestion
      - name: Set up Go
         uses: actions/setup-go@v5
         with:
           go-version-file: go.mod
           check-latest: true
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+name: "Unit Tests"
+
+on:
+  pull_request:
+    branches:
+    - main
```
Comment (joaopapereira) [inline .github/workflows/test.yml:6]:
```suggestion
    - test
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: '1.18.3'
+
+      - name: Setup test
+        run: |
+          go install github.com/onsi/ginkgo/v2/ginkgo
+
```
Comment (joaopapereira) [inline .github/workflows/test.yml:30]:
```suggestion
```

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Made some suggestions on the PR to align with main branch and also ensure that it is running for test branch

File: .github/workflows/test.yml (hunk 3 above)
Comment (gururajsh) [reply .github/workflows/test.yml:31]: I don't see any difference here or am I going blind :)

Comment (gururajsh) [reply .github/workflows/test.yml:31]: Ok.. in email I could see the difference.

**Pull request merged**